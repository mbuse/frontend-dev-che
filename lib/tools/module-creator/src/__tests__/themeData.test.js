"use strict";

const {
  initPackageJson,
  initWebpackConfigJs,
  initThemeConfigJson,
  initThemeSass,
  initPreviewSass,
  initThemeIndexJs,
  initThemeJs,
  initPreviewJs,
} = require("../themeData");

describe("initPackageJson()", () => {
  it("returns data for package.json", () => {
    expect(
      initPackageJson(
        "my-test",
        "src/js/index.js",
        {},
        {
          "some-brick": "^1.0.0",
          "some-other-brick": "^1.0.0",
        }
      )
    ).toMatchSnapshot();
  });
});

describe("initWebpackConfigJs()", () => {
  it("returns data for webpack.config.js", () => {
    expect(initWebpackConfigJs()).toMatchSnapshot();
  });
});

describe("initThemeConfigJson()", () => {
  it("returns data for theme descriptor", () => {
    expect(initThemeConfigJson("my-test", false)).toMatchSnapshot();
  });
});

describe("initThemeSass()", () => {
  it("returns data for <themeName>.sass", () => {
    expect(initThemeSass("my-test")).toMatchSnapshot();
  });
});

describe("initPreviewSass()", () => {
  it("returns data for preview.sass", () => {
    expect(initPreviewSass("my-test")).toMatchSnapshot();
  });
});

describe("initThemeIndexJs()", () => {
  it("returns data for index.js", () => {
    expect(initThemeIndexJs("my-test")).toMatchSnapshot();
  });
});

describe("initThemeJs()", () => {
  it("returns data for <themeName>.js", () => {
    expect(initThemeJs("my-test")).toMatchSnapshot();
  });
});

describe("initPreviewJs()", () => {
  it("returns data for preview.js", () => {
    expect(initPreviewJs("my-test")).toMatchSnapshot();
  });
});
