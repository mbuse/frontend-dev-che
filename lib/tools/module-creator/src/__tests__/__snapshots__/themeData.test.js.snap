// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initPackageJson() returns data for package.json 1`] = `
"{
  \\"name\\": \\"@coremedia/my-test-theme\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"build\\": \\"cross-env NODE_OPTIONS=--openssl-legacy-provider webpack\\",
    \\"deploy\\": \\"cm theme-importer deploy-theme\\",
    \\"start\\": \\"cm monitor\\",
    \\"prettier\\": \\"prettier --write \\\\\\"src/**/*.{js,json}\\\\\\"\\",
    \\"theme-importer\\": \\"cm theme-importer\\"
  },
  \\"__comment__dependencies__\\": {
    \\"__comment__\\": \\"List of all brick dependencies. In order to add a dependency, move the entry to 'dependencies'\\",
    \\"some-brick\\": \\"^1.0.0\\",
    \\"some-other-brick\\": \\"^1.0.0\\",
    \\"@coremedia/brick-utils\\": \\"^1.0.0\\",
    \\"jquery\\": \\"^3.6.0\\"
  },
  \\"dependencies\\": {
    \\"@coremedia/cm-cli\\": \\"workspace:2.0.0\\",
    \\"@coremedia/theme-utils\\": \\"workspace:3.0.0\\",
    \\"cross-env\\": \\"^7.0.3\\",
    \\"webpack\\": \\"^4.46.0\\"
  },
  \\"devDependencies\\": {
    \\"prettier\\": \\"^2.4.1\\"
  },
  \\"main\\": \\"src/js/index.js\\",
  \\"coremedia\\": {
    \\"type\\": \\"theme\\",
    \\"init\\": \\"src/js/my-test.js\\"
  },
  \\"browserslist\\": [
    \\"last 1 Chrome version\\",
    \\"last 1 Firefox version\\",
    \\"last 1 Edge version\\"
  ]
}"
`;

exports[`initPreviewJs() returns data for preview.js 1`] = `
"/*! Theme my-test: Preview JS */
// add preview specific code here...
"
`;

exports[`initPreviewSass() returns data for preview.sass 1`] = `
"/*! Theme my-test: Preview Styles */
// ### VARIABLES ###

// Own variables (need to be loaded first, so default values can be overridden)
// @see https://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default

//@import \\"variables/...\\";

// Dependency variables

@import \\"?smart-import-variables\\";

// ... add third-party dependencies here (after smart-import-variables)

// ### PARTIALS ###

// Dependency partials

// ... add third-party dependencies here (before smart-import-partials)

@import \\"?smart-import-partials\\";

// Own partials

//@import \\"partials/...\\";
"
`;

exports[`initThemeConfigJson() returns data for theme descriptor 1`] = `
"{
  \\"name\\": \\"my-test\\",
  \\"styles\\": [
    {
      \\"type\\": \\"webpack\\",
      \\"src\\": \\"src/sass/my-test.scss\\"
    }
  ],
  \\"scripts\\": [
    {
      \\"type\\": \\"webpack\\",
      \\"src\\": \\"src/js/my-test.js\\"
    }
  ],
  \\"l10n\\": {
    \\"bundleEncoding\\": \\"UTF-8\\",
    \\"bundleNames\\": [
      \\"My-test\\"
    ]
  }
}"
`;

exports[`initThemeIndexJs() returns data for index.js 1`] = `
"/*! Theme my-test */
import \\"./my-test.js\\";
"
`;

exports[`initThemeJs() returns data for <themeName>.js 1`] = `
"//import $ from \\"jquery\\";
//import * as utils from \\"@coremedia/brick-utils\\";

// --- JQUERY DOCUMENT READY -------------------------------------------------------------------------------------------

//$(function () {
//  utils.log(\\"Theme my-test is used.\\");
//});
"
`;

exports[`initThemeSass() returns data for <themeName>.sass 1`] = `
"/*! Theme my-test */
// ### VARIABLES ###

// Own variables (need to be loaded first, so default values can be overridden)
// @see http://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default

@import \\"variables\\";

// ### PARTIALS ###

@import \\"partials\\";
"
`;

exports[`initWebpackConfigJs() returns data for webpack.config.js 1`] = `
"const { webpackConfig } = require(\\"@coremedia/theme-utils\\");

module.exports = webpackConfig;"
`;
